BLSQ/iaso
# IA-3061 IA-2853 IA-3217 Public Registry

Alpha Version of the public registry.
See notes below as this require some configuration to be created via Django Admin

Related JIRA tickets : IA-3061 IA-2853 IA-3217

## Self proofreading checklist


## Changes

Creation of a `/api/public/registry/` endpoint which return a registry config, and the instances and instance detail with the values redacted according to the corresponding Public Registry Config (Still very basic at the moment)

There are also some modification on DataSource model to have a `public` flag and also to the APIs so that when a datasource is public you can access its data without being logged in.

Also this includes all the changes made in the frontend to allow for the Public Registry (Setup in a _registry_ plugin)

## Documentation

See `docs/pages/dev/reference/public_registry/public_registry.md`

## How to test

You need to create a public registry config via Django Admin. For the moment the registry slug **MUST** be `default_registry` as it is hardcoded in the frontend. In the next version this should be passed to the frontend via URL parameter.
You need to set the used data source as public via django admin.
Then you can visit `/registry/` while being UNLOGGED from the site.
Then click on the "Explorer les donnees" button and you should see the public registry page which is very similar to the non public registry.


## Print screen / video

<img width="1127" alt="Screenshot 2024-07-18 at 12 37 08" src="https://github.com/user-attachments/assets/ec4ff1d7-96c3-4502-a64c-22fafaab1a1e">
